### ISSUE TITLE
Should alpaka override BOOST_COMP_... variables?

### ISSUE THREAD

**Issue by sbastrakov** [alpaka-group/alpaka#1072]
After figuring this out while working on #1070 , I am wondering if our practice of overriding Boost.Predef variables is that good. In case alpaka is used with other code that makes use of such variables (or uses some libraries that do) it can lead to inconsistencies that are hard to track.

Is there anything wrong with e.g. instead of undefining `BOOST_COMP_NVCC` and defining our own, define another variable like `ALPAKA_COMP_NVCC` and using it inside alpaka. Then it would be more clear when we mean Boost variables and when their alpaka "improvements", and also avoid potential risks of changing the logic of users' code without them really knowing.

**Comment by BenjaminW3** [comment]
Our goal should not be to override Boost.Predef variables.
My goal when I introduced `Boost.Predef` instead of hand-written macros was to get rid of a lot of boiler plate.
However, it turned out that `Boost.Predef` was missing some features (PTX architecture, clang CUDA compiler, ...) or had broken detection for some compilers (nvcc, intel, ...). Therefore we had to conditionally override some of the macros.
The goal was always to port back our fixes into `Boost.Predef` so that we can get rid of the workarounds later.
